server/src/main/java/io/crate/execution/engine/aggregation/impl/TopKAggregation.java

```diff
@@ -484,6 +486,7 @@ private static long toLong(DataType<?> type, Object o) {
             case LongType.ID -> (Long) o;
             case DoubleType.ID -> NumericUtils.doubleToSortableLong((Double) o);
             case FloatType.ID -> (long) NumericUtils.floatToSortableInt((Float) o);
+            case IntegerType.ID -> ((Integer) o).longValue();
```

Comment:
Note this applies only for the non-doc-values case where the int is already an integer. In the doc-value case this is never called.